In [3]:
from array import *

In [5]:
value = array("i",[5,-6,7,8,-9,2,3])
print(value)

array('i', [5, -6, 7, 8, -9, 2, 3])


In [7]:
value = array("i",[5,6,7,9.77,2,3])
print(value)

TypeError: integer argument expected, got float

In [8]:
value = array("I",[5,6,7,90,2,3])
print(value)

array('I', [5, 6, 7, 90, 2, 3])


In [9]:
value = array("i",[-5,-6,7,90,2,3])
print(value.buffer_info())

(1905839671792, 6)


#buffer_info() will give size of an array
#("1905839671792" is yhe address , "6" is the size)

In [10]:
value = array("I",[5,6,7,90,2,3])
print(value[2])

7


In [11]:
value = array("I",[5,6,7,90,2,3])
print(len(value))

6


In [13]:
value = array("u",['a','u','u','h'])
for e in value:
    print(e)

a
u
u
h


In [15]:
array_num = array('i', [1,3,5,7,9,10,15])
array_num.byteswap()
#array.byteswap()
#(“Byteswap” all items of the array. It is useful when reading data from a file written on a machine
#with a different byte order. This is only supported for values which are 1, 2, 4, or 8 bytes in size.
#For other types of values, RuntimeError is raised.)

In [16]:
array_num = array('i', [1,3,5,7,9])
array_num.typecode

'i'

In [21]:
from array import *
vals=array("i", [2,3,4,5,6,7,8])
newArr=array(vals.typecode, (a for a in vals))
for e in newArr:
    print(e)

2
3
4
5
6
7
8


#new array bnaya "vals.typecode" becoz we dont know the type of array, "a for a in vals" means take one-one value from
#previous array

In [22]:
from array import *
vals=array("i", [2,3,4,5,6,7,8])
newArr=array(vals.typecode, (a*a for a in vals))
for e in newArr:
    print(e)

4
9
16
25
36
49
64


## Array values from User in Python

In [25]:
from array import *

In [30]:
arr = array("i",[])
n=int(input("enter the length of the array = "))
for i in range(n):
    x=int(input("enter the next number = "))
    arr.append(x)
print(arr)

enter the length of the array = 5
enter the next number = 9
enter the next number = 8
enter the next number = 7
enter the next number = 6
enter the next number = 5
array('i', [9, 8, 7, 6, 5])


In [1]:
from array import *
arr=array("i",[1,2,3],[4,5,6])
print(arr)

TypeError: array() takes at most 2 arguments (3 given)

In [ ]:
#array module works only on 1D.

In [18]:
from array import *
vals = array('i',[1,2,3,4])
vals.reverse()
print(vals)

array('i', [4, 3, 2, 1])


In [22]:
from array import *
val = array('i',[1,2,3,4])
print(val.index(2))

1


#### miscellaneous

In [4]:
import array as arr

In [24]:
yo = arr.array("i",[-1,2,3,4,5])

In [25]:
print(yo)

array('i', [-1, 2, 3, 4, 5])


In [26]:
print(yo.buffer_info())

(1752474051856, 5)


In [27]:
yo.typecode

'i'

In [28]:
yo.reverse()
print(yo)

array('i', [5, 4, 3, 2, -1])


In [29]:
yo[0]

5

In [32]:
yo[-1]=90
print(yo)

array('i', [5, 4, 3, 2, 90])


In [35]:
len(yo)

5

In [44]:
for i in range(len(yo)):
    print(yo[i])

5
4
3
2
90
